pandas/core/indexes/base.py

```diff
@@ -5194,7 +5194,7 @@ def _from_join_target(self, result: np.ndarray) -> ArrayLike:
         """
         if isinstance(self.values, BaseMaskedArray):
             return type(self.values)(result, np.zeros(result.shape, dtype=np.bool_))
-        elif isinstance(self.values, ArrowExtensionArray):
+        elif isinstance(self.values, (ArrowExtensionArray, StringArray)):
```

Comment:
just out of interest, why can't this be done for all EAs?

Reply:
We enhanced this when we added ``_get_join_target``. Previously we returned result unchanged, so not sure if this would potentially break stuff

Reply:
ok - this looks fine to be, but would prefer if someone more familiar with core/indexes could take a look too